src/decorators/delay_node.cpp

```diff
@@ -37,19 +36,45 @@ NodeStatus DelayNode::tick()
 
     if (!delay_started_)
     {
+        delay_complete = false;
         delay_started_ = true;
         setStatus(NodeStatus::RUNNING);
         if (msec_ > 0)
         {
-            std::lock_guard<std::timed_mutex> l(delay_mutex);
-            std::thread t(&DelayNode::delay, this);
-            t.join();
+            timer_id_ = timer_.add(std::chrono::milliseconds(msec_), [this](bool aborted) {
+                std::unique_lock<std::mutex> lk(delay_mutex);
+                if (!aborted)
+                {
+                    delay_complete = true;
+                }
+                else
+                {
+                    delay_aborted = true;
+                }
+            });
         }
```

Comment:
add an `else` and set `delay_complete = true;` Otherwise this will delay for-ever.. If this is intended add it to your small example. Never the less, add the expected behavior for numbers <= 0 as an description ? 🙂

Reply:
Actually this is intended since we have only two cases here (I dont see any third case here which can cause the node to get stuck in delay mode) :
1.  If the timer is executed cleanly aborted flag will be false hence we will set delay_complete and the node will tick the child
2. If the timer is aborted then aborted flag will be true and we will set delay_aborted as true which will cause the delay to return failure.

Reply:
In the latest commit I have handled the case of numbers<0 as a runtime error, hope this solves the query